torchrl/objectives/ppo.py

```diff
@@ -434,11 +434,14 @@ def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
             ess = (2 * lw.logsumexp(0) - (2 * lw).logsumexp(0)).exp()
             batch = log_weight.shape[0]
 
-        if not advantage.shape == log_weight.shape:
-            raise RuntimeError(
-                f"advantage.shape and log_weight.shape do not match (got {advantage.shape} "
-                f"and {log_weight.shape})"
-            )
+        if advantage.shape != log_weight.shape:
+            try:
+                log_weight = log_weight.view_as(advantage)
+            except Exception as err:
+                raise RuntimeError(
+                    f"advantage.shape and log_weight.shape do not match (got {advantage.shape} "
+                    f"and {log_weight.shape})"
+                ) from err
```

Comment:
I would drop this but change the indexing in the model as I indicated in https://github.com/tcbegley/rl/pull/34 [:, [-1], :] -> [:, -1, :]
like that we would not have to change PPO